DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskInterception.swift

```diff
@@ -65,6 +67,10 @@ public class URLSessionTaskInterception {
         )
     }
 
+    public func enableRUM2APM() {
+        self.rum2APM = true
+    }
```

Comment:
suggestion/ This API looks very odd (_"set bool flag"_). IIRC, this is set from RUM to skip further work in Tracing. Because we use `x-datadog-origin: rum` header for the same purpose (to let APM backend know that trace is managed by RUM) - have we considered using it here, instead of this new flag-API?

According to:
```diff
- /// To make sure the generated traces from RUM don’t affect APM Index Spans counts.
- /// **Note:** it is only added to requests that we create RUM Resource for (it is not injected when RUM feature is disabled and only Tracing is used). WDYT @maxep?
- public static let ddOrigin = (field: "x-datadog-origin", value: "rum")
```
if `request` was intercepted by RUM it will have `x-datadog-origin: rum` set. I can imagine that checking this header occurrence could be equivalent to `if rum2APM` check in Tracing handler - without introducing new API and only using existing constructs. WDYT @maxep ?

Reply:
Totally, it's even better to use the origin as it is set at the very beginning of the interception.

Reply:
It works for `URLRequest` but not for `URL` instrumentation as we cannot inject header into a URL object 😭

Reply:
@ncreated I've changed the `rum2APM: Bool` to a `origin: String?`, this value is set automatically by reading the HTTP header field  "x-datadog-origin" and by RUM when starting an interception. LMKWYT!